tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -714,6 +715,47 @@ public:
    void AcceptVisitor(Detail::RFieldVisitor &visitor) const final;
 };
 
+/// The generic field an std::bitset<N>. All compilers we care about store the bits in an array of unsigned long.
+/// TODO(jblomer): reading and writing efficiency should be improved; currently it is one bit at a time
+/// with an array of bools on the page level.
+class RBitsetField : public Detail::RFieldBase {
+protected:
+   std::size_t fN;
+
+protected:
+   std::unique_ptr<Detail::RFieldBase> CloneImpl(std::string_view newName) const final
+   {
+      return std::make_unique<RBitsetField>(newName, fN);
+   }
+   const RColumnRepresentations &GetColumnRepresentations() const final;
+   void GenerateColumnsImpl() final;
+   void GenerateColumnsImpl(const RNTupleDescriptor &desc) final;
+   std::size_t AppendImpl(const Detail::RFieldValue &value) final;
+   void ReadGlobalImpl(NTupleSize_t globalIndex, Detail::RFieldValue *value) final;
+
+public:
+   RBitsetField(std::string_view fieldName, std::size_t N);
+   RBitsetField(RBitsetField &&other) = default;
+   RBitsetField &operator=(RBitsetField &&other) = default;
+   ~RBitsetField() override = default;
+
+   using Detail::RFieldBase::GenerateValue;
+   Detail::RFieldValue GenerateValue(void *where) final
+   {
+      memset(where, 0, GetValueSize());
+      return Detail::RFieldValue(true /* captureFlag */, this, where);
+   }
+   Detail::RFieldValue CaptureValue(void *where) final
+   {
+      return Detail::RFieldValue(true /* captureFlag */, this, where);
+   }
+   size_t GetValueSize() const final { return (fN + sizeof(unsigned long) * 8 - 1) / 8; }
```

Comment:
This looks a bit off, as probably the value size should be an integral multiple of `sizeof(unsigned long)`.  FWIW, libstdc++ computes this as follows:
```c++
#define _GLIBCXX_BITSET_BITS_PER_WORD  (__CHAR_BIT__ * __SIZEOF_LONG__)
#define _GLIBCXX_BITSET_WORDS(__n) \
  ((__n) / _GLIBCXX_BITSET_BITS_PER_WORD + \
   ((__n) % _GLIBCXX_BITSET_BITS_PER_WORD == 0 ? 0 : 1))
```